gitops/generate.go

```diff
@@ -0,0 +1,204 @@
+//
+// Copyright 2021-2022 Red Hat, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package gitops
+
+import (
+	routev1 "github.com/openshift/api/route/v1"
+	appstudiov1alpha1 "github.com/redhat-appstudio/application-service/api/v1alpha1"
+	"github.com/redhat-appstudio/application-service/gitops/resources"
+	"github.com/spf13/afero"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+
+	yaml "github.com/redhat-appstudio/application-service/gitops/yaml"
+)
+
+const (
+	deploymentFileName = "deployment.yaml"
+	serviceFileName    = "service.yaml"
+	routeFileName      = "route.yaml"
+)
+
+// Generate takes in a given Component CR and
+// spits out a deployment, service, and route file to disk
+func generate(fs afero.Fs, outputFolder string, component appstudiov1alpha1.Component) {
+	deployment := generateDeployment(component)
+
+	k := resources.Kustomization{}
+	k.AddResources(deploymentFileName)
+	resources := map[string]interface{}{
+		deploymentFileName: deployment,
+	}
+
+	// If a targetPort was specified, also generate a service and route
+	if component.Spec.TargetPort != 0 {
+		service := generateService(component)
+		route := generateRoute(component)
+		k.AddResources(deploymentFileName, serviceFileName, routeFileName)
+		resources[serviceFileName] = service
+		resources[routeFileName] = route
+	}
+	resources["kustomization.yaml"] = k
+
+	yaml.WriteResources(fs, outputFolder, resources)
```

Comment:
probably should check the returned err here

Reply:
Good catch, fixed.